.github/workflows/todo-auto-issues.yml

```diff
@@ -0,0 +1,48 @@
+name: Check for TODOs
+
+on:
+  pull_request:
+    types:
+      - closed
+    branches:
+      - main
+
+jobs:
+  scan-for-todos:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged == true
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        ref: ${{ github.event.pull_request.merge_commit_sha }}
+    
+    - name: Scan for TODOs
+      id: todos
+      run: |
+        git fetch origin main
+        todos=$(git diff ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.merge_commit_sha }} --unified=0 | awk '/^\+.*TODO:/ { sub(/^.*TODO:\s*/, ""); sub(/^ */, ""); $0=toupper(substr($0,1,1)) substr($0,2); print $0 }')
+        
+        # Set output
+        echo "todo_list=$todos" >> $GITHUB_OUTPUT;
+
+    - name: Create GitHub issue for each TODO
+      if: steps.todos.outputs.todo_list != ''
+      run: |
+        # Split the `todo_list` string into an array
+        SAVEIFS=$IFS   # Save current IFS (Internal Field Separator)
+        IFS=$'\n'      # Change IFS to newline char
+        todos=("${{ steps.todos.outputs.todo_list }}") # split the `todo_list` string into an array by the same name
+        IFS=$SAVEIFS   # Restore original IFS
```

Comment:
note: this is just a way of converting `todo_list` from string with newlines to an array so we could iterate over todos